From f67ef896bdbd9a95c6d291fd0f6fe254ecbd8d50 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Mon, 19 Aug 2024 08:55:31 +0200
Subject: [PATCH] Play audio attachments within the application

Audio enclosures can be played in the same way how videos are played.
---
 src/enclosure_button.rs | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/enclosure_button.rs b/src/enclosure_button.rs
index c88d5486..a8acac9c 100644
--- a/src/enclosure_button.rs
+++ b/src/enclosure_button.rs
@@ -110,7 +110,7 @@ impl EnclosureButton {
                 ImageDialog::new_url(&enclosure.article_id, &enclosure.url)
                     .present(Some(&App::default().main_window()));
                 return;
-            } else if mime.starts_with("video") {
+            } else if mime.starts_with("video") || mime.starts_with("audio") {
                 App::default().main_window().play_enclosure(enclosure);
                 return;
             }
@@ -149,7 +149,7 @@ impl EnclosureButton {
         } else if mime.starts_with("video") {
             i18n("Open Attached Video")
         } else if mime.starts_with("audio") {
-            i18n("Watch Youtube Video")
+            i18n("Open Attached Audio")
         } else {
             i18n("Open Attachment")
         }
-- 
GitLab

